src/QAT/qat/purr/backends/verification.py

```diff
@@ -89,25 +93,38 @@ def get_verification_model(qpu_type: QPUVersion) -> Optional[VerificationModel]:
         )
 
     if qpu_type.make == Lucy.__name__:
-        # TODO: Should have an apply_setup_to_hardware in live which people can use to build our own architecture.
-        model = VerificationModel(qpu_type, LucyVerificationEngine)
-        raise NotImplementedError("No lucy-specific model created yet.")
+        model = get_default_lucy_hardware()
+    else:
+        raise NotImplementedError(f"No specific model created yet for {qpu_type}.")
 
-    return None
+    return model
 
 
 class VerificationEngine(QuantumExecutionEngine, ABC):
+
+    def __init__(self, model):
+        super().__init__(model)
+
     def _execute_on_hardware(self, sweep_iterator, package: QatFile):
         veri_list = list(package.meta_instructions)
         veri_list.extend(package.instructions)
         self.verify_instructions(veri_list)
 
     @abstractmethod
-    def verify_instructions(self, instructions: List[Instruction]):
+    def verify_instructions(self, instructions: List[Instruction], max_circuit_duration: int = 90000):
```

Comment:
Revert, as mentioned circuit duration is based very explicitly on hardware version.

Reply:
Done